operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -569,21 +606,61 @@ func (dc *CassandraDatacenter) GetConfigAsJSON() (string, error) {
 	return modelParsed.String(), nil
 }
 
+// Gets the defined CQL port for NodePort.
+// 0 will be returned if NodePort is not configured.
+// The SSL port will be returned if it is defined,
+// otherwise the normal CQL port will be used.
+func (dc *CassandraDatacenter) GetNodePortCqlPort() int {
+	if !dc.IsNodePortEnabled() {
+		return 0
+	}
+
+	if dc.Spec.Networking.NodePort.CqlSsl != 0 {
+		return dc.Spec.Networking.NodePort.CqlSsl
+	} else if dc.Spec.Networking.NodePort.Cql != 0 {
+		return dc.Spec.Networking.NodePort.Cql
+	} else {
+		return 9042
+	}
+}
+
+// Gets the defined broadcast/intranode port for NodePort.
+// 0 will be returned if NodePort is not configured.
+// The SSL port will be returned if it is defined,
+// otherwise the normal broadcast port will be used.
+func (dc *CassandraDatacenter) GetNodePortBroadcastPort() int {
+	if !dc.IsNodePortEnabled() {
+		return 0
+	}
+
+	if dc.Spec.Networking.NodePort.BroadcastSsl != 0 {
```

Comment:
The port fallback logic is a bit squirrelly and went through multiple designs via slack/standup conversations.  What I have landed on is nodeport ssl ports take first priority, then nodeport non-ssl ports, then the defaults of 9042 for native and 7000 for broadcast.  If the nodeport native/cql port is defined, then the native port will be omitted from the normal service.